pkg/scheduler/api/types.go

```diff
@@ -309,6 +309,15 @@ type HostPriority struct {
 	Score int
 }
 
+// FeatureDependencies provides the feature and its dependent predicates and priorities.
+type FeatureDependency struct {
+	Name                  string   // Name of the feature gate.
+	NeededPredicateList   []string // List of predicates needed if we enable this predicate or priority
+	NeededPriorityList    []string // List of priorities needed if we enable this predicate or priority.
+	ExcludedPredicateList []string // List of predicates that should be excluded if we enable this predicate/priority.
+	ExcludedPriorityList  []string // List of priorities that should be excluded if we enable this predicate/priority
+}
```

Comment:
Maybe change `[]string` to `map[string]struct{}` so searching will be more performant - from O(n) to O(1). And also avoid duplication.

Reply:
Yes, ofcourse, that is what I meant here - https://github.com/kubernetes/kubernetes/pull/68103/files#r214220679 and in the TODO: